sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -557,23 +557,30 @@ object ViewHelper {
   private def collectTemporaryObjects(
       catalog: SessionCatalog, child: LogicalPlan): (Seq[Seq[String]], Seq[String]) = {
     def collectTempViews(child: LogicalPlan): Seq[Seq[String]] = {
-      child.collect {
+      child.flatMap {
```

Comment:
@peter-toth, thanks for fixing this.
any particular reason to replace `collect` to `flatMap`?

Reply:
I have the same comment, too. IMO the previous one looks okay.

Reply:
seems like it simplifies `collect(...).flatten` to `flatMap(...)`

Reply:
as @cloud-fan explained